EIPS/eip-erc20-swiss-compl.md

```diff
@@ -0,0 +1,182 @@
+---
+eip: <to be assigned>
+title: ERC-20-compatible Swiss Compliant Asset Token
+author: Gianluca Perletti (@Perlets9), Alan Scarpellini (@alanscarpellini), Roberto Gorini (@robertogorini), Manuel Olivi (@manvel79)
+discussions-to: Pull Request
+status: Draft
+type: Standards Track
+category: ERC
+created: 2020-09-08
+requires: 20
+---
+
+## Simple Summary
+
+An interface for asset tokens (or security tokens), compliant with Swiss Law and compatible with ERC-20.
+
+## Abstract
+
+This new standard is an ERC-20 compatible token with restrictions that comply with one or more one the following Swiss laws: Stock Exchange Act, the Banking Act, the Financial Market Infrastructure Act, the Act on Collective Investment Schemes and the Anti-Money Laundering Act. The Financial Services Act and the Financial Institutions Act must also be considered. The solution achieved meet also the European jurisdiction.
+
+This new standard meets the new era of asset tokens (or security tokens). These new methods manage securities ownership during issuance and trading. The issuer is the only role that can manage a white-listing and the only one that is allowed to execute “freeze” or “revoke” functions.
+
+## Motivation
+
+In its ICO guidance dated February 16, 2018, FINMA (Swiss Financial Market Supervisory Authority) defines asset tokens as tokens representing assets and/or relative rights. It explicitly mentions that asset tokens are analogous to and can economically represent shares, bonds, or derivatives. The long list of relevant financial market laws mentioned above reveal that we need more methods than with Payment and Utility Token.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+The words "asset tokens" and "security tokens" can be considered synonymous.
+
+Every ERC-toBeAssigned compliant contract must implement the ERCtoBeAssigned interface.
+
+### ERCtoBeAssigned (Token Contract)
+
+``` solidity
+interface ERCtoBeAssigned {
+  
+  /// @dev This emits when funds are reassigned
+  event FundsReassigned(address from, address to, uint256 amount);
+
+  /// @dev This emits when funds are revoked
+  event FundsRevoked(address from, uint256 amount);
+
+  /// @dev This emits when an address is frozen
+  event FundsFrozen(address target);
+
+  /**
+  * @dev Transfer tokens from a specified address to another one
+  * this operation can be performmed only by an Issuer.
+  * @param _from The address from which the tokens are withdrawn
+  * @param _to The address that receives the tokens
+  * @return true if the tokens are transferred
+  */
+  function reassign(address _from, address _to) external;
+
```

Comment:
```suggestion
```

Any functions that are issuer only shouldn't be part of the specification since there isn't a many-to-many relationship between tokens and users.  Essentially, this is an implementation detail and not part of the standardized interface.

In this case, for any given token there will be one issuer and that issuer can simply use tools that work with that specific token.  For example, if one token names this function `reassign` and another token names this function `steal`, both tokens can work and general purpose tools can still work with them.  Each issuer would simply use the tooling for that specific token, or design the token around a particular tool that they like.